### PR TITLE
Improve dev guides

### PR DESCRIPTION
**Pull request by kamilmielnik**
- Update docs to the current state of the app

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30823)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "repository": "https://github.com/metabase/metabase",
   "license": "private",
   "engines": {
-    "node": ">=16",
+    "node": "^16",
```
Comment (kamilmielnik) [inline package.json:9]: I assumed this is only for informative purposes.
Please let me know if this change should be done in a separate PR.

**Review by kulyk — APPROVED** [review]

**Review by ranquild — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (uladzimirdev) [reply package.json:9]: should be good

**Review by uladzimirdev — APPROVED** [review]

**Review by iethree — APPROVED** [review]

**Review by losrebellos — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (kamilmielnik) [reply package.json:9]: It looks like it's breaking [some tests](https://github.com/metabase/metabase/actions/runs/5010843765/jobs/8981087216?pr=30823) though:

> Run yarn build-static-viz
> yarn run v[1](https://github.com/metabase/metabase/actions/runs/5010843765/jobs/8981087216?pr=30823#step:4:1).22.19
> error metabase@0.0.0: The engine "node" is incompatible with this module. Expected version "^16". Got "1[8](https://github.com/metabase/metabase/actions/runs/5010843765/jobs/8981087216?pr=30823#step:4:9).16.0"
> error Commands cannot run with an incompatible environment.
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> Error: Process completed with exit code 1.

I'll revert this line and make sure this PR is only about markdown updates.

**Comment by uladzimirdev** [comment]
That’s weird. Maybe we need to prioritize upgrade to node 18 or setup node
16 at ci

Чт, 18 мая 2023 г. в 09:21, Kamil Mielnik ***@***.***>:
